ghcide/src/Development/IDE/Core/FileExists.hs

```diff
@@ -106,11 +107,11 @@ getFileExistsMapUntracked = do
   return v
 
 -- | Modify the global store of file exists.
-modifyFileExists :: IdeState -> [(NormalizedFilePath, FileChangeType)] -> IO ()
+modifyFileExists :: IdeState -> [(NormalizedFilePath, FileChangeType)] -> IO [Key]
```

Comment:
Can you please extend the haddock to make it clear what the returned `[Key]` means?
This doesn't make it clear what should be done with these returned keys, so I'd maybe add a note-style comment about this somewhere stating what problem these extra IO [Key] arguments/return types mean and how they are used.